gst/mqtt/mqttsrc.c

```diff
@@ -1100,8 +1138,9 @@ cb_mqtt_on_connect (void *context, MQTTAsync_successData * response)
 
   self->mqtt_respn_opts.subscribeOptions.retainHandling = 1;
   /** @todo Support QoS option */
-  ret = MQTTAsync_subscribe (self->mqtt_client_handle, self->mqtt_topic, 0,
-      &self->mqtt_respn_opts);
+  ret =
+      MQTTAsync_subscribe (self->mqtt_client_handle, self->mqtt_topic,
+      self->mqtt_qos, &self->mqtt_respn_opts);
```

Comment:
Did you test with multiple subscribers? In specific, you need to check the case when one can get the message but the other one fails.

Reply:
I'm testing in various ways but still didn't understand the exact mechanism yet.